From: Michael Olbrich <m.olbrich@pengutronix.de>
Subject: add configure option for kernel header dir

The keymap extra generates code based on linux/input.h. The header file
used during code generation must be the the one used when compiling.
By default /usr/include/linux/input.h is used for code generation.

This patch adds a configure option to specify the directory containing the
kernel headers.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

---
 Makefile.am  |    2 +-
 configure.ac |    5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

Index: b/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -565,7 +565,7 @@ udevkeymapforcereldir = $(libexecdir)/ke
 dist_udevkeymapforcerel_DATA = \
 	extras/keymap/keymaps/force-release/samsung-other
 
-extras/keymap/keys.txt: /usr/include/linux/input.h
+extras/keymap/keys.txt: $(KERNEL_HEADERS_DIR)/linux/input.h
 	mkdir -p extras/keymap
 	$(AM_V_GEN)$(AWK) '/^#define.*KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' < $< | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
 
Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -75,6 +75,11 @@ AC_ARG_ENABLE([modem-modeswitch],
 	[], [enable_modem_modeswitch=yes])
 AM_CONDITIONAL([ENABLE_MODEM_MODESWITCH], [test "x$enable_modem_modeswitch" = xyes])
 
+AC_ARG_WITH([kernel-headers-dir],
+	AS_HELP_STRING([--with-kernel-headers-dir=DIR], [where to find the kernel headers]),
+	[kernel_headers=${withval}], [kernel_headers=/usr/include])
+AC_SUBST([KERNEL_HEADERS_DIR], [$kernel_headers])
+
 if test "x$enable_keymap" = xyes; then
 	AC_PATH_PROG([GPERF], [gperf])
 	if test -z "$GPERF"; then
